projects/04_cube_xr/main.cpp

```diff
@@ -0,0 +1,373 @@
+            // position          // vertex colors
+            -1.0f,-1.0f,-1.0f,   1.0f, 0.0f, 0.0f,  // -Z side
+             1.0f, 1.0f,-1.0f,   1.0f, 0.0f, 0.0f,
+             1.0f,-1.0f,-1.0f,   1.0f, 0.0f, 0.0f,
+            -1.0f,-1.0f,-1.0f,   1.0f, 0.0f, 0.0f,
+            -1.0f, 1.0f,-1.0f,   1.0f, 0.0f, 0.0f,
+             1.0f, 1.0f,-1.0f,   1.0f, 0.0f, 0.0f,
+
+            -1.0f, 1.0f, 1.0f,   0.0f, 1.0f, 0.0f,  // +Z side
+            -1.0f,-1.0f, 1.0f,   0.0f, 1.0f, 0.0f,
+             1.0f, 1.0f, 1.0f,   0.0f, 1.0f, 0.0f,
+            -1.0f,-1.0f, 1.0f,   0.0f, 1.0f, 0.0f,
+             1.0f,-1.0f, 1.0f,   0.0f, 1.0f, 0.0f,
+             1.0f, 1.0f, 1.0f,   0.0f, 1.0f, 0.0f,
+
+            -1.0f,-1.0f,-1.0f,   0.0f, 0.0f, 1.0f,  // -X side
+            -1.0f,-1.0f, 1.0f,   0.0f, 0.0f, 1.0f,
+            -1.0f, 1.0f, 1.0f,   0.0f, 0.0f, 1.0f,
+            -1.0f, 1.0f, 1.0f,   0.0f, 0.0f, 1.0f,
+            -1.0f, 1.0f,-1.0f,   0.0f, 0.0f, 1.0f,
+            -1.0f,-1.0f,-1.0f,   0.0f, 0.0f, 1.0f,
+
+             1.0f, 1.0f,-1.0f,   1.0f, 1.0f, 0.0f,  // +X side
+             1.0f, 1.0f, 1.0f,   1.0f, 1.0f, 0.0f,
+             1.0f,-1.0f, 1.0f,   1.0f, 1.0f, 0.0f,
+             1.0f,-1.0f, 1.0f,   1.0f, 1.0f, 0.0f,
+             1.0f,-1.0f,-1.0f,   1.0f, 1.0f, 0.0f,
+             1.0f, 1.0f,-1.0f,   1.0f, 1.0f, 0.0f,
+
+            -1.0f,-1.0f,-1.0f,   1.0f, 0.0f, 1.0f,  // -Y side
+             1.0f,-1.0f,-1.0f,   1.0f, 0.0f, 1.0f,
+             1.0f,-1.0f, 1.0f,   1.0f, 0.0f, 1.0f,
+            -1.0f,-1.0f,-1.0f,   1.0f, 0.0f, 1.0f,
+             1.0f,-1.0f, 1.0f,   1.0f, 0.0f, 1.0f,
+            -1.0f,-1.0f, 1.0f,   1.0f, 0.0f, 1.0f,
+
+            -1.0f, 1.0f,-1.0f,   0.0f, 1.0f, 1.0f,  // +Y side
+            -1.0f, 1.0f, 1.0f,   0.0f, 1.0f, 1.0f,
+             1.0f, 1.0f, 1.0f,   0.0f, 1.0f, 1.0f,
+            -1.0f, 1.0f,-1.0f,   0.0f, 1.0f, 1.0f,
+             1.0f, 1.0f, 1.0f,   0.0f, 1.0f, 1.0f,
+             1.0f, 1.0f,-1.0f,   0.0f, 1.0f, 1.0f,
+        };
+        // clang-format on
+        uint32_t dataSize = ppx::SizeInBytesU32(vertexData);
+
+        grfx::BufferCreateInfo bufferCreateInfo       = {};
+        bufferCreateInfo.size                         = dataSize;
+        bufferCreateInfo.usageFlags.bits.vertexBuffer = true;
+        bufferCreateInfo.memoryUsage                  = grfx::MEMORY_USAGE_CPU_TO_GPU;
+
+        PPX_CHECKED_CALL(GetDevice()->CreateBuffer(&bufferCreateInfo, &mVertexBuffer));
+
+        void* pAddr = nullptr;
+        PPX_CHECKED_CALL(mVertexBuffer->MapMemory(0, &pAddr));
+        memcpy(pAddr, vertexData.data(), dataSize);
+        mVertexBuffer->UnmapMemory();
+    }
+
+    // Viewport and scissor rect
+    mViewport    = {0, 0, float(GetWindowWidth()), float(GetWindowHeight()), 0, 1};
+    mScissorRect = {0, 0, GetWindowWidth(), GetWindowHeight()};
+}
+
+void ProjApp::Render()
+{
+    PerFrame& frame = mPerFrame[0];
+
+    uint32_t currentViewIndex = 0;
+#if defined(PPX_BUILD_XR)
```

Comment:
I still prefer to leave the PPX_BUILD_XR for these 2 examples for now to be clear of where the required changes are for OpenXR integration so that it would be easier for people to understand. Later when we have the final interface, we could do a cleaning round to remove those PPX_BUILD_XR if necessary.

Reply:
Would it be of interest to build them with `PPX_BUILD_DIR=FALSE` then? So this path doesn't rot?

Reply:
removed all `PPX_BUILD_XR` as suggested